firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionDetailsViewController.swift

```diff
@@ -0,0 +1,417 @@
+        image.layer.masksToBounds = true
+        image.layer.cornerRadius = 5
+    }
+    private let connectionStatusLabel: UILabel = .build { label in
+        label.font = FXFontStyles.Regular.subheadline.scaledFont()
+        label.numberOfLines = 0
+        label.adjustsFontForContentSizeCategory = true
+        label.textAlignment = .left
+    }
+    private let dividerView: UIView = .build { _ in }
+    private var lockImageHeightConstraint: NSLayoutConstraint?
+
+    // MARK: Verified By View
+    private let verifiedByView: UIView = .build { view in
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.layer.cornerRadius = TPMenuUX.UX.viewCornerRadius
+        view.layer.maskedCorners = [.layerMinXMaxYCorner, .layerMaxXMaxYCorner]
+        view.layer.masksToBounds = true
+    }
+    private let verifiedByLabel: UILabel = .build { label in
+        label.font = FXFontStyles.Regular.subheadline.scaledFont()
+        label.numberOfLines = 0
+        label.adjustsFontForContentSizeCategory = true
+        label.textAlignment = .left
+    }
+
+    // MARK: See Certificates View
+    private lazy var viewCertificatesButton: LinkButton = .build { button in
+        button.titleLabel?.textAlignment = .left
+        button.titleLabel?.numberOfLines = 0
+        button.addTarget(self, action: #selector(self.viewCertificatesTapped), for: .touchUpInside)
+    }
+
+    // MARK: - Variables
+
+    private var constraints = [NSLayoutConstraint]()
+    var viewModel: TrackingProtectionDetailsModel
+    var notificationCenter: NotificationProtocol
+    var themeManager: ThemeManager
+    var themeObserver: NSObjectProtocol?
+    let windowUUID: WindowUUID
+    var currentWindowUUID: UUID? { return windowUUID }
+
+    // MARK: - View Lifecycle
+
+    init(with viewModel: TrackingProtectionDetailsModel,
+         windowUUID: WindowUUID,
+         and notificationCenter: NotificationProtocol = NotificationCenter.default,
+         themeManager: ThemeManager = AppContainer.shared.resolve()) {
+        self.viewModel = viewModel
+        self.windowUUID = windowUUID
+        self.notificationCenter = notificationCenter
+        self.themeManager = themeManager
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    deinit {
+        notificationCenter.removeObserver(self)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        updateViewDetails()
+        listenForThemeChange(view)
+        applyTheme()
+    }
+
+    private func setupView() {
+        constraints.removeAll()
+
+        setupContentView()
+        setupHeaderView()
+        setupConnectionStatusView()
+        setupVerifiedByView()
+        setupSeeCertificatesView()
```

Comment:
similar to the comment above, if these can be in separate classes / files, that would be great!

Reply:
@petruSt24 why not add these in a stack view, I feel that it cleans up the extra constraints